config.go

```diff
@@ -1,28 +1,85 @@
+)
+
+var ErrInvalidConfig = errors.New("invalid configuration")
 
 // gRPC Client Configuration.
 type Config struct {
-	Address          string            `json:"address"`
-	Token            string            `json:"token"`
-	TenantID         string            `json:"tenant_id"`
-	APIKey           string            `json:"api_key"`
-	ClientCertPath   string            `json:"client_cert_path"`
-	ClientKeyPath    string            `json:"client_key_path"`
-	CACertPath       string            `json:"ca_cert_path"`
-	TimeoutInSeconds int               `json:"timeout_in_seconds"`
-	Insecure         bool              `json:"insecure"`
-	Headers          map[string]string `json:"headers"`
+	// Address of the service to connect to.
+	//
+	// Address is typically in the form "hostname:port" but may also be a
+	// a Unix socket or DNS URI.
+	// See https://github.com/grpc/grpc/blob/master/doc/naming.md#name-syntax for more details.
+	Address string `json:"address"`
+
+	// A JWT to be used for authentication with the service.
+	//
+	// Token and APIKey are mutually exclusive.
+	Token string `json:"token"`
+
+	// An API key to be used for authentication with the service.
+	APIKey string `json:"api_key"`
+
+	// An Aserto tenant ID.
+	TenantID string `json:"tenant_id"`
+
+	// An aserto account ID.
+	AccountID string `json:"account_id"`
+
+	// In mTLS connections, ClientCertPath is the path of the client's
+	// certificate file.
+	ClientCertPath string `json:"client_cert_path"`
+
+	// In mTLS connections, ClientKeyPath is the path of the client's
+	// private key file.
+	ClientKeyPath string `json:"client_key_path"`
+
+	// In TLS connections, CACertPath is the path of a CA certificate to
+	// validate the server's certificate against.
+	CACertPath string `json:"ca_cert_path"`
+
+	// In TLS connections, skip verification of the server certificate.
+	Insecure bool `json:"insecure"`
+
+	// Disable TLS and use a plaintext connection.
+	NoTLS bool `json:"no_tls"`
+
+	// Additional headers to include in requests to the service.
+	Headers map[string]string `json:"headers"`
 }
 
-func (cfg *Config) ToConnectionOptions(dop DialOptionsProvider) ([]ConnectionOption, error) {
-	options := []ConnectionOption{
-		WithInsecure(cfg.Insecure),
+// Connects to the service specified in Config, possibly with additional
+// connection options.
+func (cfg *Config) Connect(opts ...ConnectionOption) (*grpc.ClientConn, error) {
+	connOpts := &ConnectionOptions{Config: *cfg}
+	if err := connOpts.Apply(opts...); err != nil {
+		return nil, err
 	}
 
+	return Connect(connOpts)
+}
+
+// Converts the Config into a ConnectionOption slice that can be passed to NewConnection().
+func (cfg *Config) ToConnectionOptions() ([]ConnectionOption, error) {
 	if cfg.APIKey != "" && cfg.Token != "" {
-		return nil, errors.New("both api_key and token are set")
+		return nil, errors.Wrap(ErrInvalidConfig, "api_key and token are mutually exclusive")
+	}
+
+	if cfg.Insecure && cfg.NoTLS {
```

Comment:
Do we need a mutually exclusive check for mTLS and (insecure | NoTLS)?

Reply:
Good catch!